src/EditorFeatures/CSharpTest/SplitComment/SplitCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,139 @@
+            }
+            view.SetMultiSelection(snapshotSpans);
+
+            var undoHistoryRegistry = workspace.GetService<ITextUndoHistoryRegistry>();
+            var commandHandler = new SplitCommentCommandHandler(
+                undoHistoryRegistry,
+                workspace.GetService<IEditorOperationsFactoryService>());
+
+            if (!commandHandler.ExecuteCommand(new ReturnKeyCommandArgs(view, view.TextBuffer), TestCommandExecutionContext.Create()))
+            {
+                callback();
+            }
+
+            if (expectedOutputMarkup != null)
+            {
+                MarkupTestFile.GetSpans(expectedOutputMarkup,
+                    out var expectedOutput, out ImmutableArray<TextSpan> expectedSpans);
+
+                Assert.Equal(expectedOutput, view.TextBuffer.CurrentSnapshot.AsText().ToString());
+
+                if (verifyUndo)
+                {
+                    // Ensure that after undo we go back to where we were to begin with.
+                    var history = undoHistoryRegistry.GetHistory(document.GetTextBuffer());
+                    history.Undo(count: originalSelections.Count);
+
+                    var currentSnapshot = document.GetTextBuffer().CurrentSnapshot;
+                    Assert.Equal(originalSnapshot.GetText(), currentSnapshot.GetText());
+                }
+            }
+        }
+
+        /// <summary>
+        /// verifyUndo is needed because of https://github.com/dotnet/roslyn/issues/28033
+        /// Most tests will continue to verifyUndo, but select tests will skip it due to
+        /// this known test infrastructure issure. This bug does not represent a product
+        /// failure.
+        /// </summary>
+        private void TestHandled(
+            string inputMarkup, string expectedOutputMarkup,
+            bool verifyUndo = true, IndentStyle indentStyle = IndentStyle.Smart)
+        {
+            TestWorker(
+                inputMarkup, expectedOutputMarkup,
+                callback: () =>
+                {
+                    Assert.True(false, "Should not reach here.");
+                },
+                verifyUndo, indentStyle);
+        }
+
+        private void TestNotHandled(string inputMarkup)
+        {
+            var notHandled = false;
+            TestWorker(
+                inputMarkup, null,
+                callback: () =>
+                {
+                    notHandled = true;
+                });
+
+            Assert.True(notHandled);
+        }
+
+
+        [WorkItem(38516, "https://github.com/dotnet/roslyn/issues/38516")]
+        [WpfFact, Trait(Traits.Feature, Traits.Features.SplitComment)]
+        public void TestSplitComment()
+        {
+            TestHandled(
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test [||]Comment
+    }
+}",
+@"public class Program
+{
+    public static void Main(string[] args) 
+    { 
+        // Test 
+//Comment
```

Comment:
this does not appear to be desirable behavior.

Reply:
I still need to fix this. I was having trouble with the indentation. I think it’s due to acting on trivia instead of a statement